### PR TITLE
Remove jQuery from CoBlocks

### LINKED ISSUE #1365: ISBAT use CoBlocks without jQuery
### Is your feature request related to a problem? Please describe.
CoBlocks should consider removing jQuery as a required dependency for the plugin. Getting a look at the plugin I find that 20 files reference `jQuery` which could be refactored to use plain JavaScript. This change would help to make CoBlocks more performant. 

## Required changes
Events & Post Carousel block
- [ ] Remove Slick carousel in favor of Flickity. (Slick is a jQuery extension) - Will be removed following Events/Post-Carousel refactor
- [ ] `src/js/coblocks-events.js` initializes slick in the DOM. - Needs Block refactor
- [ ] `src/js/coblocks-slick-initializer-front.js` - Will be removed following Events/Post-Carousel refactor ( In progress )
- [ ] `src/js/coblocks-slick-initializer.js` - Will be removed following Events/Post-Carousel refactor ( In progress )

Form block - jQuery UI Datepicker | Should be another open-source date picker.
- [x] Remove jQuery from Form block `edit.js` file ( #1376 )
- [ ] `src/js/coblocks-datepicker.js` Should be removed. ( #2180  )
- [ ] `includes/class-coblocks-form.php` Should not reference jquery-ui-datepicker. ( #2180 )
- [x] `src/js/coblocks-google-recaptcha.js` Should not use jQuery. ( #2131 )

Gif block
- [x] Remove jQuery from Gif `edit.js` file ( #1376 )

Gallery Carousel `src/js/coblocks-accordion-carousel.js` - ( PR: https://github.com/godaddy-wordpress/coblocks/pull/2147 )
  - [x] Refactor to vanilla-js swiper package
  
Gallery Masonry `src/js/coblocks-masonry.js`
  - [ ] Use vanilla JS to initialize Masonry. (https://github.com/desandro/masonry#initialize)

Google Maps Script `src/js/coblocks-google-maps.js`
- [x] Refactor script ( #2130 )

CoBlocks Lightbox `src/js/coblocks-lightbox.js`
- [x] Refactor script ( https://github.com/godaddy-wordpress/coblocks/pull/1429 ) 

CoBlocks Crop Settings Extension & Component
- [x] Remove jQuery from Crop Settings extension and component ( #1376 ) 

**Cleanup tasks**
- [ ] Review `includes/class-coblocks-block-assets.php` and remove jQuery from dependency arrays.  
       - This file will be able to remove all references to jQuery once the following blocks are refactored
           - Events
           - Post-Carousel 
- [ ] Review package.json and remove jQuery from `devDependencies`
      - As stated above, both the Events and Post-Carousel blocks must be refactored away from Slick before we can remove jQuery from the devDependencies file

**Inspired by WordPress.org support request**
https://wordpress.org/support/topic/did-you-start-using-jquery-in-a-recent-release/

### PR DESCRIPTION
**Pull request by olafleur-godaddy**
### Description
Remove the remains of jQuery that are not needed anymore.

Please tell me if I broke something or if I forgot to remove something :)

Fixes #1365 

This is the culmination of the work of a lot of folks, especially @AnthonyLedesma and @snovosel :)

### CODE REVIEW

**Review by jonathanbardo — APPROVED** [review]
@olafleur-godaddy this seems to work as expected on my test CoBlocks page. I think this will probably need another pair of eyes though before merging.

File: includes/class-coblocks-block-assets.php
Comment (AnthonyLedesma) [inline includes/class-coblocks-block-assets.php:364]: I think we still need this because the v2 Masonry block is not compatible with WordPress version below 5.7.2. We allow users to use both versions of the block using a special wrapper and we need to keep the old logic around for that purpose.

**Comment by AnthonyLedesma** [comment]
Please allow me to give this a good test with everything around the Masonry block and the v1-v2 wrapper that it uses before we merge this.

File: includes/class-coblocks-block-assets.php
Comment (AnthonyLedesma) [reply includes/class-coblocks-block-assets.php:364]: No, we are good on this I suppose. Turns out the Masonry block broke with 5.6 somewhere along the way. Let us go ahead and get this in and then we will have to look at that Masonry block later.

**Review by AnthonyLedesma — APPROVED** [review]

**Pull request merged**
